<a href="https://colab.research.google.com/github/IyinoluwaDon/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/IyinoluwaDon/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

Selected Lane: Refresh / Content Opportunity Scoring

Reason: Content decay is a major problem for content teams. The starter dataset tracks content metadata and performance trends over time, making it ideal for creating a prioritized queue of pages that need updating.

In [2]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.



## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

Decision: Deciding which decaying content items to rewrite during the upcoming sprint.

Action: Produce a prioritized list of pages with recommendations for the editorial team.

Cost of a wrong call: False positive wastes editing budget on good pages; false negative ignores decaying pages and loses organic traffic.

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [3]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


import os
import pandas as pd

if not os.path.exists("flyrank-ml-internship"):
    !git clone https://github.com/IyinoluwaDon/flyrank-ml-internship.git

df = pd.read_csv("flyrank-ml-internship/data/raw/content_refresh_anonymized.csv")
print(f"Total Rows: {len(df)}")
print(f"Total Columns: {len(df.columns)}")
display(df.describe())

Cloning into 'flyrank-ml-internship'...
remote: Enumerating objects: 114, done.
remote: Counting objects: 100% (114/114), done.
remote: Compressing objects: 100% (70/70), done.
remote: Total 114 (delta 28), reused 101 (delta 28), pack-reused 0 (from 0)
Receiving objects: 100% (114/114), 1.84 MiB | 9.26 MiB/s, done.
Resolving deltas: 100% (28/28), done.
Total Rows: 30000
Total Columns: 44


,search_volume,competition,cpc,word_count,char_count,impressions_90d,clicks_90d,pageviews_90d,sessions_90d,users_90d,...,sessions_prev_30d,content_age_days,age_tier_order,days_since_last_update,ctr,avg_position,engagement_rate,scroll_rate,ai_traffic_pct,trend_pct
count,27532.000000,27532.000000,27532.000000,22301.000000,22301.000000,30000.000000,30000.000000,30000.000000,30000.000000,30000.000000,...,30000.000000,30000.00000,30000.000000,30000.000000,30000.000000,30000.00000,30000.000000,29875.000000,30000.000000,26612.000000
mean,158.882391,0.146954,0.485342,3107.760325,20665.277835,5200.366300,16.097333,49.942467,37.066633,35.937700,...,10.283000,256.16780,4.786533,46.098300,0.510733,16.34238,2.534520,18.212921,0.768196,-4.785969
std,1518.270825,0.285241,2.101560,1452.382598,10115.344042,16838.019547,75.076958,152.101430,107.069131,103.748185,...,42.578003,132.70793,0.790392,42.078709,3.279162,15.21679,8.310096,29.472768,7.429454,473.861780
min,0.000000,0.000000,0.000000,8.000000,40.000000,1.000000,0.000000,0.000000,1.000000,1.000000,...,0.000000,90.00000,3.000000,1.000000,0.000000,0.00000,0.000000,0.000000,0.000000,-100.000000
25%,0.000000,0.000000,0.000000,2413.000000,15644.000000,81.000000,0.000000,2.000000,2.000000,2.000000,...,1.000000,132.00000,4.000000,20.000000,0.000000,6.20000,0.000000,0.000000,0.000000,-62.600000
50%,10.000000,0.000000,0.000000,2877.000000,19116.000000,731.000000,1.000000,8.000000,7.000000,7.000000,...,2.000000,236.00000,5.000000,20.000000,0.070000,10.80000,0.000000,5.000000,0.000000,-33.500000
75%,20.000000,0.130000,0.000000,3666.000000,24011.000000,3615.250000,7.000000,33.000000,27.000000,27.000000,...,7.000000,333.00000,5.000000,104.000000,0.290000,22.30000,1.350000,23.530000,0.000000,0.000000
max,74000.000000,1.000000,100.360000,9546.000000,111158.000000,517715.000000,4178.000000,5998.000000,4345.000000,4913.000000,...,4247.000000,564.00000,6.000000,373.000000,100.000000,245.00000,100.000000,300.000000,300.000000,44900.000000


1. Dataset Volume: The dataset contains 30,000 total content items evaluated across 44 features/metrics.

2. Widespread Traffic Decay: Over half of all pages show declining traffic trends, with a median trend_pct drop of -33.5% (mean -4.79%).

3. Stale Content Profile: Content items have an average age (content_age_days) of 256.2 days (~8.5 months), generating a low average of 16.1 clicks per 90 days (clicks_90d), proving a high volume of pages need intervention.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

What I CAN claim: I can claim empirical, statistical prioritization of content items exhibiting observable traffic decay based on measured metrics (e.g., negative trend percentages, low CTR).

What I CANNOT claim: I cannot claim direct causal proof of search engine algorithmic penalties or guarantee a specific traffic volume increase after an update.

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## Self-check

Before you submit, confirm each line honestly:

[x] Every section above is filled — markdown thinking AND the code that backs it

[x] The notebook runs top to bottom with no errors (Runtime $\rightarrow$ Run all)

[x] No client names, URLs, or private queries anywhere

[x] My claims use careful words: observed, measured, directional, decision-support

[x] Committed to my repo under work/notebooks/ — then submit your repo URL on the card.